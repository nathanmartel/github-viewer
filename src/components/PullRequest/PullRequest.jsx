import React from 'react';
import PropTypes from 'prop-types';

const PullRequest = ({ number, title, state, url }) => (
  <>
    <br />
    <a href={url}>({state}) - #{number} - {title}</a>
  </>
);

PullRequest.propTypes = {
  id: PropTypes.number.isRequired,
  number: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  state: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default PullRequest;
