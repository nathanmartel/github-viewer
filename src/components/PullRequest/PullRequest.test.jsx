import React from 'react';
import { shallow } from 'enzyme';
import PullRequest from './PullRequest';

describe('PullRequest component', () => {
  it('renders PullRequest', () => {
    const PR = {
      id: 1,
      number: 3,
      title: 'Add thing',
      state: 'open',
      url: 'http://www.hello.com'
    };
    const wrapper = shallow(<PullRequest {...PR} />);
    expect(wrapper).toMatchSnapshot();
  });
});
