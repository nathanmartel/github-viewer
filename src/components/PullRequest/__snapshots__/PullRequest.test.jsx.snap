// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequest component renders PullRequest 1`] = `
<a
  href="http://www.hello.com"
>
  (
  open
  ) - #
  3
   - 
  Add thing
</a>
`;
